File: src/main/java/io/github/cheatbook/amzrevenuemanager/domain/entity/Settlement.java
package io.github.cheatbook.amzrevenuemanager.domain.entity;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.IdClass;
import jakarta.persistence.Table;
import jakarta.persistence.Transient;
import lombok.Data;

/**
 * 決済エンティティです。
 */
@Entity
@Data
@Table(name = "settlement")
@IdClass(SettlementId.class)
public class Settlement {

    /**
     * 決済ID
     */
    @Id
    @Column(length = 30)
    private String settlementId;

    /**
     * トランザクションの種類
     */
    @Column(length = 50)
    private String transactionType;

    /**
     * 注文ID
     */
    @Id
    @Column(length = 50)
    private String orderId;

    /**
     * 注文商品コード
     */
    @Id
    private Long orderItemCode;

    /**
     * 金額の種類
     */
    @Column(length = 50)
    private String amountType;

    /**
     * 金額の説明
     */
    @Id
    @Column(length = 50)
    private String amountDescription;

    /**
     * 金額
     */
    private BigDecimal amount;

    /**
     * 計上日時
     */
    @Id
    private LocalDateTime postedDateTime;

    /**
     * SKU
     */
    @Column(length = 50)
    private String sku;

    /**
     * 購入数量
     */
    private Integer quantityPurchased;

    /**
     * 親SKU（DBテーブル外）
     */
    @Transient
    private String parentSku;

    /**
     * 親SKUを取得します。
     *
     * @return 親SKU
     */
    public String getParentSku() {
        return this.parentSku;
    }

    /**
     * 親SKUを設定します。
     *
     * @param parentSku 親SKU
     */
    public void setParentSku(String parentSku) {
        this.parentSku = parentSku;
    }
}

File: backend-typescript-lambda/src/domain/models/Settlement.ts
/**
 * 決済情報インターフェース
 */
export interface Settlement {
  /** 決済ID */
  settlementId: string;
  /** 行ID (一意性を確保するための連番) */
  lineId: string;
  /** トランザクションの種類 (Order, Refund など) */
  transactionType: string;
  /** 注文ID */
  orderId: string;
  /** Amazon注文ID */
  amazonOrderId: string;
  /** 注文商品コード */
  orderItemCode: string;
  /** 金額の種類 (ItemPrice, ItemFees など) */
  amountType: string;
  /** 金額の説明 (Principal, Commission など) */
  amountDescription: string;
  /** 金額 */
  amount: number;
  /** 計上日時 */
  postedDateTime: string;
  /** SKU */
  sku: string;
  /** 購入数量 */
  quantityPurchased: number;
}
